sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3645,15 +3646,14 @@ class AstBuilder extends DataTypeAstBuilder
    * converted into "helloworld".
    *
    * Special characters can be escaped by using Hive/C-style escaping.
+   *
+   * Note: With the modified stringLit grammar rule, visitStringLit now returns a Token
+   * that already represents the coalesced value of multiple literals.
    */
   private def createString(ctx: StringLiteralContext): String = {
-    if (conf.escapedStringLiterals) {
-      ctx.stringLit.asScala.map(x => stringWithoutUnescape(visitStringLit(x))).mkString
-    } else if (conf.getConf(LEGACY_CONSECUTIVE_STRING_LITERALS)) {
```

Comment:
With this change, we now ignore the `LEGACY_CONSECUTIVE_STRING_LITERALS` config [1]. Is this intended?

[1] https://github.com/apache/spark/blob/eb117a642bcd217b81f6ac4980f2030fcf17d46b/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L4382-L4390__